### PR TITLE
RWA-1647-db-resilience-check

### PR DESCRIPTION
**Pull request by serdaroguzhanakin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-1647

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    void should_handle_database_outage_and_log_issue_when_getting_all_messages_in_new_state() {
+        doThrow(new JDBCConnectionException("An error occurred when getting all message in new state", null))
+            .when(caseEventMessageRepository)
+            .getAllMessagesInNewState();
+
+        await().ignoreException(Exception.class)
+            .pollInterval(5, SECONDS)
+            .atMost(60, SECONDS)
+            .untilAsserted(() -> {
+                messageReadinessConsumer.run();
```
Comment (rasikasri) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java:89]: When you call messageReadinessConsumer.run() here, you are not testing actual application behaviour. This should be called by the executor automatically. Basically test should pass without this line.

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                //The purpose of the test: we should see retry when an error occurred.
+                //So after 2 trying test will pass
+                if (count.get() > 1) {
+                    isDone.set(true);
+                }
+                count.getAndIncrement();
+                assertTrue(isDone.get());
```
Comment (rasikasri) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java:96]: Can't you simplify this with assertTrue(count.get() > 1). Or using untilTrue isntead of untilAsserted

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        doThrow(new TransactionTimedOutException("An error occurred when updating message state"))
+            .when(caseEventMessageRepository)
+            .updateMessageState(any(), any());
+
+
+        await().ignoreException(Exception.class)
+            .pollInterval(5, SECONDS)
+            .atMost(60, SECONDS)
+            .untilAsserted(() -> {
+                messageReadinessConsumer.run();
```
Comment (rasikasri) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java:119]: Same comments applies here.

**Review by rasikasri — CHANGES_REQUESTED** [review]

**Review by sullivangareth — APPROVED** [review]

**Comment by sullivangareth** [comment]
Approved, but agree with changes suggested by Rasika

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java (hunk 1 above)
Comment (serdaroguzhanakin) [reply src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java:89]: you are right but it makes this more readable

Comment (serdaroguzhanakin) [reply src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java]: done

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java (hunk 2 above)
Comment (serdaroguzhanakin) [reply src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java]: done

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java (hunk 3 above)
Comment (serdaroguzhanakin) [reply src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java]: done

**Review by sullivangareth — APPROVED** [review]

**Review by rasikasri — APPROVED** [review]

**Pull request merged**